src/wasm/wasm-ir-builder.cpp

```diff
@@ -151,9 +151,20 @@ void IRBuilder::push(Expression* expr) {
 
   applyDebugLoc(expr);
   if (binaryPos && func && lastBinaryPos != *binaryPos) {
-    func->expressionLocations[expr] =
-      BinaryLocations::Span{BinaryLocation(lastBinaryPos - codeSectionOffset),
-                            BinaryLocation(*binaryPos - codeSectionOffset)};
+    auto start = BinaryLocation(lastBinaryPos - codeSectionOffset);
+    auto end = BinaryLocation(*binaryPos - codeSectionOffset);
+    // Some expressions already have their start noted, and we are just seeing
+    // their last segment (like an Else).
+    auto iter = func->expressionLocations.find(expr);
```

Comment:
Let's always do just one lookup by using `.insert()` here to get an iterator, then fixing up just the `end` if an entry already exists.

Reply:
Done.